### PR TITLE
Use pre-built ONNX Runtime binaries.

### PR DESCRIPTION
**Pull request by mc-nv**
server: https://github.com/triton-inference-server/server/pull/7717

### CODE REVIEW

**Review by nv-kmcgill53 — APPROVED** [review]

File: cmake/download_onnxruntime.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  if(NOT DOWNLOAD_RESULT EQUAL 0)
+    message(NOTICE "Failed to download: ${TRITON_ONNXRUNTIME_PACKAGE_URL}")
+  else()
+    message(NOTICE "Download successful: ${DOWNLOAD_PATH}" )
+
+    file(ARCHIVE_EXTRACT INPUT ${DOWNLOAD_PATH} DESTINATION ${EXTRACT_DIR} VERBOSE )
+
+    set(TRITON_ONNXRUNTIME_INCLUDE_PATHS ${EXTRACT_DIR}/include)
+    set(TRITON_ONNXRUNTIME_LIB_PATHS ${EXTRACT_DIR}/lib)
```
Comment (fpetrini15) [inline cmake/download_onnxruntime.cmake:21]: Is this going to break the RHEL build?

Comment (mc-nv) [reply cmake/download_onnxruntime.cmake:21]: No unless you start feed the prebuilt asset for RHEL.

Comment (fpetrini15) [reply cmake/download_onnxruntime.cmake:21]: We may end up using this for our Linux builds in the future, but I suppose we can enhance it when needed.

Comment (mc-nv) [reply cmake/download_onnxruntime.cmake:21]: Is it mean that we can resolve this thread?

**Review by fpetrini15 — APPROVED** [review]

File: cmake/download_onnxruntime.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
+if(DEFINED TRITON_ONNXRUNTIME_PACKAGE_URL)
+
+  set(DOWNLOAD_PATH "${CMAKE_BINARY_DIR}/_deps/downloads/onnxruntime.zip")
```
Comment (GuanLuo) [inline cmake/download_onnxruntime.cmake:3]: How do we get the original `onnxruntime.zip`? Is that a separate (optional) build step in our pipeline?

File: cmake/download_onnxruntime.cmake [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  # file(DOWNLOAD ... STATUS DOWNLOAD_STATUS) returns a list with 2 elements
+  list(GET DOWNLOAD_STATUS 0 DOWNLOAD_RESULT)
+
+  if(NOT DOWNLOAD_RESULT EQUAL 0)
+    message(NOTICE "Failed to download: ${TRITON_ONNXRUNTIME_PACKAGE_URL}")
+  else()
+    message(NOTICE "Download successful: ${DOWNLOAD_PATH}" )
+
+    file(ARCHIVE_EXTRACT INPUT ${DOWNLOAD_PATH} DESTINATION ${EXTRACT_DIR} VERBOSE )
```
Comment (GuanLuo) [inline cmake/download_onnxruntime.cmake:18]: Do we care about the ORT version? If so, perhaps we should package the version in a file in the compressed artifact and check if the version matches?

File: cmake/download_onnxruntime.cmake (hunk 2 above)
Comment (mc-nv) [reply cmake/download_onnxruntime.cmake:3]: It's manual for now.

File: cmake/download_onnxruntime.cmake (hunk 3 above)
Comment (mc-nv) [reply cmake/download_onnxruntime.cmake:18]: We do care about version. 
Version is compressed. 
I can add validator if it's your ask, please raise a ticket against me.

Comment (mc-nv) [reply cmake/download_onnxruntime.cmake:18]: We can put a hack like this: 
```CMake 
        file(READ "${EXTRACT_DIR}/VERSION_NUMBER" DOWNLOADED_ONNXRUNTIME_VERSION)
        if( NOT ${DOWNLOADED_ONNXRUNTIME_VERSION} VERSION_EQUAL ${TRITON_BUILD_ONNXRUNTIME_VERSION})
          set(TRITON_ONNXRUNTIME_INCLUDE_PATHS ${EXTRACT_DIR}/include)
          set(TRITON_ONNXRUNTIME_LIB_PATHS ${EXTRACT_DIR}/lib)
        endif()
```

Comment (GuanLuo) [reply cmake/download_onnxruntime.cmake:18]: I think this is sufficient for now

Comment (mc-nv) [reply cmake/download_onnxruntime.cmake:18]: I don't believe that I understand you comment in right, does it mean that we can close the thread? 
Or you willing me to extend PR with proposed above code snippet?

Comment (GuanLuo) [reply cmake/download_onnxruntime.cmake:18]: Would be great if you can extend PR with proposed above code snippet

**Review by GuanLuo — APPROVED** [review]

**Pull request merged**